Add tests for safeMethodNameFor:basedOn: and revisit it

clacker opened the pull request:
TestCase for the safeMethodNameFor:basedOn: method of the ReAbstractTransformation class.

In response to issue #11870.

clacker committed:
A TestCase for the safeMethodNameFor:basedOn: method of the AbstractTransformation class.
In responce to issue #11870.

Ducasse commented:
I think that the tests show that the logic of safeMethodNameFor:basedOn: is bogus. 
So this is good. 

Because for binary selector it should not add the number at the end since these are not syntactically correct. 

Here is what I would do. Either

- check explicitly in safeMethodNameFor:basedOn: if the argument represents a binary selector (using isBinary) and in the case just return it. 
- rename safeMethodNameFor:basedOn: into safeAccessorNameFor:basedOn:
- Update the corresponding test.

clacker committed:
now correctly handles binary messages.

clacker commented:
I changed safeMethodNameFor:basedOn: to explicitly check for a binary selector, just returning the selector in that case.  I modified the test cases to yield the correct results for binary selectors. It failed the check for "**Archive JUnit-formatted test results**," is that something I need to change something in my branch to fix?
